Add facility field changes to contributor notification emails

jwalgran opened the pull request:
## Overview

Add a list of updated field values to contributor email notifications.

Connects #635

## Testing Instructions

* Claim a facility and approve the claim.
* Browse the facility profile page at http://localhost:6543/claimed/{id}/ and change some fields:
  * Update `Facility Name` to "Changed Name"
  * Update `Phone Number` to "Changed Phone Number"
  * Turn OFF the `Publicly Visible` switch next to `Phone Number`
  * Update `Website` to "change.org"
* Restart `./scripts/server` to ensure that emails are displayed in the console.
* Save the changes. Verify that the email printed to the console includes the following. (Verify both the plain text and HTML versions)

```
The following details have been changed:


  - Facility name: Changed Name
  - Facility website: change.org
```
* Change all the field values on the form, turning on all publicly visible switches.
* Save the form and verify that the email printed to the console shows all the field changes (Verify both the plain text and HTML versions)

```
The following details have been changed:


  - Description: Changed Description 2
  - Facility name: Changed Name 2
  - Address: Changed Address 2
  - Facility phone number: Changed Phone 2
  - Facility website: change-2.org
  - Minimum order quantity: Changed Quantity 2
  - Average lead time: Changed Lead 2
  - Contact person: Changed Contact 2
  - Contact email: changed-email@example.com
  - Office official name: Changed Office 2
  - Office address: Changed Office Address 2
  - Office country code: DM
  - Office phone number: Changed Office Phone 2
  - Parent company: Brand E
```


## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

jwalgran commented:
This PR includes changing one instance of `.update` to `.save` which partially resolves #648.

kellyi commented:
Heads up that there is a merge conflict in one of the email template files.

jwalgran commented:
Rebased on develop and resolved the conflict.

kellyi commented on src/django/api/models.py:
```diff
@@ -406,142 +408,241 @@ class FacilityClaim(models.Model):
     verification_method = models.TextField(
         null=False,
         blank=True,
+        verbose_name='Verification method',
```
Very trivial but I just read that the Django doc recommend not capitalizing the first letter of these verbose name fields since Django will capitalize them as it needs: https://docs.djangoproject.com/en/2.2/topics/db/models/#verbose-field-names

However, if it doesn't matter feel free to ignore this.

jwalgran replied:
I will update these to be consistent with the Django docs and your new fields in #650. I will also make the required update to make sure that we do the capitalization when preparing the email context object.

jwalgran replied:
Done in fixups 55bd009, 8932d49, and 227fd9a

kellyi commented on src/django/api/models.py:
```diff
@@ -406,142 +408,241 @@ class FacilityClaim(models.Model):
+        'office_country_code',
+        'office_phone_number',
+        'parent_company',
+    )
```
#650 will have a few new fields to add to this list, but it should be easy to adjust.

jwalgran commented:
I agree, and will hold on this until #650 is done so I can incorporate the additional fields.

jwalgran commented:
I retested this after rebasing and adding support for the new fields added in #650.

I changed all the fields on a claim but did not enable any switches. I got this expected list in the email:

```
- Description: change desc
- Facility name in English: change name
- Facility name in native language: change nâme native
- Address: change addres
- Minimum order quantity: change min
- Average lead time: change lead
- Facility workers count: 1
- Percentage of female workers: 1
- Facility type: Cut and Sew / RMG
- Parent company: Auditor A
```

I switched on the public fields and changed their values and got this expected list in the email:

```
- Facility phone number: change phone 2
- Facility website: change2.web
- Contact person: change contact 2
- Contact email: change.contact.2@email.com
- Office official name: change office 2
- Office address: change office address 2
- Office country code: AX
- Office phone number: change office phone 2
```

jwalgran commented:
Thanks for the review. Will merge after CI passes.
